src/interpreter.rs

```diff
@@ -372,13 +372,15 @@ fn is_low_der_signature(vch_sig: &ValType) -> Result<bool, ScriptError> {
     // https://bitcoin.stackexchange.com/a/12556:
     //     Also note that inside transaction signatures, an extra hashtype byte
     //     follows the actual signature data.
-    let vch_sig_copy = vch_sig.clone();
+    let (_, vch_sig_copy) = vch_sig.split_last().unwrap_or_else(|| {
+        unreachable!("`is_valid_signature_encoding` checks that the length is at least 9.")
+    });
```

Comment:
```suggestion
    let (_, vch_sig_copy) = vch_sig
        .split_last()
        .expect("`is_valid_signature_encoding` has checked that the length is at least 9");
```
In general `.unwrap_or_else(|| unreachable!(msg))` should instead be `.expect(msg)`. The latter means the same thing, but is shorter. (Don't worry about rustfmt sometimes making code _look_ more verbose.)